### ISSUE TITLE
RISCV32gc code gen for if (a > b && c > d) has too many xori

### ISSUE THREAD

**Issue by apinski-cavium** [llvm/llvm-project#57320]
```c
int g(void);

int f(int a, int b, int c, int d)
{
  if (a > b && c > d)
    return g();
  return 1;
}
```
Produces:
```asm
f:                                      # @f
        slt     a0, a1, a0
        xori    a0, a0, 1
        slt     a1, a3, a2
        xori    a1, a1, 1
        or      a0, a0, a1
        bnez    a0, .LBB0_2
        tail    g
.LBB0_2:
        li      a0, 1
        ret
```
But the `xori` is not needed if `beqz` is used instead. Even if it was needed, there should have been only one as it can be gone after the or instead.

**Comment by topperc** [comment]
I have a patch I’m working on for this.

**Comment by apinski-cavium** [comment]
I should mention I noticed it while looking into GCC code generation too:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=106724
So I thought I will file the LLVM too.

**Comment by topperc** [comment]
Should be fixed in ec91d761ac03378456fe0528e0fe8ea4dd70927c
